### PR TITLE
Add LogConfigurator, and add logging for CSV export

### PR DESCRIPTION
**Pull request by DiggidyDave**
### CATEGORY

Choose one

- [ ] Bug Fix
- [X] Enhancement (new features, refinement)
- [ ] Refactor
- [X] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

- Added analytic event logging for CSV exports (including num rows, database info, etc).
- Added `LOG_CONFIGURATOR` config option, and `DefaultLogConfigurator` as default to keep current behavior. This allows local installation config to add handlers etc to take action based on logged events.

### TEST PLAN
I tested locally, and there is a test to exercise the basic mechanism.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@betodealmeida @mistercrunch

### CODE REVIEW

File: tests/logging_configurator_tests.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# licensed to the apache software foundation (asf) under one
```
Comment (etr2460) [inline tests/logging_configurator_tests.py]: this is using all lowercase, will probably fail the license check

File: superset/utils/logging_configurator.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                logging.StreamHandler()
+            )  # pylint: disable=no-member
+            superset_logger.setLevel(logging.INFO)  # pylint: disable=no-member
+
+        logging.getLogger("pyhive.presto").setLevel(logging.INFO)  # noqa: T484
+
+        logging.basicConfig(format=app_config.get("LOG_FORMAT"))  # noqa: T484
+        logging.getLogger().setLevel(app_config.get("LOG_LEVEL"))  # noqa: T484
+
+        if app_config.get("SILENCE_FAB"):
```
Comment (etr2460) [inline superset/utils/logging_configurator.py]: looks like this is being done twice?

File: tests/logging_configurator_tests.py (hunk 1 above)
Comment (DiggidyDave) [reply tests/logging_configurator_tests.py]: weird, I copied from other files... I wonder if something in pre-commit did that? Oh well, good catch. Thanks. Fixed.

File: superset/utils/logging_configurator.py (hunk 2 above)
Comment (DiggidyDave) [reply superset/utils/logging_configurator.py]: Thx. fixed.

**Review by xtinec — APPROVED** [review]
LGTM

**Pull request merged**